### DISCUSSION TITLE
Cypress component testing with state adapt store has name collisions with fast running tests

### DISCUSSION THREAD

**Discussion by kingjordan** [state-adapt/state-adapt#85] [Q&A] [answered]
Hello,

I have an **Angular 17** app where I am using **cypress** **component** testing extensively for developing.  Obviously some of the app components need to access the store to test actions/sources etc..

The problem I am facing is if I have a test file that runs multiple tests in a a row and each test uses the defaultStoreProvider I start getting naming collisions like the following:

> (uncaught exception)Error: Path 'app.auth' collides with 'app.auth', which has already been initialized as a state path.

Say I have 15 tests for the component, only about 3 of these will fail.  However, If i run then individually they all pass, I think the speed of the tests is messing with the browser redux dev tools plugin connected to state adapt.  

here is an example of my code:
```typescript
  it("renders empty sections 'add button' when sections.length = 0", () => {
    cy.viewport(700, 900);
    cy.mount(SessionComponent, {
      providers: [defaultStoreProvider],
      componentProperties: {
        session$: of(defaultSessionNoSections),
      },
    })
      .get('[cy-data="empty-section-button-add"]')
      .should("exist");
  });
```

Does anyone have any ideas how I can deal with this issue?

**Comment by mfp22** [comment, marked as answer]
Uh oh. I've seen this on StackBlitz before. When you edit a file, sometimes it refreshes in a way where the new page somehow loads before the last one was completely destroyed. This might require a creative solution, but the first thing to try would be to use getId() to get a unique path for the store. `path: 'app.auth' + getId()`

**Reply by mfp22** [reply]
It's not ideal, but it's at least something to try for now.

**Reply by kingjordan** [reply]
thanks, I used ``` path: 'app.auth' + new Date().getTime().toString(),``` to get a unique value. but getId() is cleaner! Now the tests run fine.  Is there a way to provide a store and ignore this error?

**Reply by mfp22** [reply]
Date.now() is shorter than new Date().getTime()

Rather than using defaultStoreProvider? Try https://state-adapt.github.io/angular/docs/angular#providestore

**Reply by kingjordan** [reply]
is there an option when defining a custom store config to ignore naming collision errors?

**Reply by mfp22** [reply]
No. I wouldn't be throwing an error if it wasn't critical. The new store would have its state nuked by the old store.

**Reply by mfp22** [reply]
I've spent probably 10 hours trying to figure out a way around this by the way. Something would have to change over time, something fundamental. I don't know what.

**Reply by kingjordan** [reply]
I totally understand, and the getID is an easy way to handle this, so for me it is all good!  Thanks for all your hard work, I am really enjoying trying to embrace declarative programming.
